Server/mods/deathmatch/logic/luadefs/CLuaGenericDefs.cpp

```diff
@@ -26,3 +38,153 @@ bool CLuaGenericDefs::IsTransferBoxVisible()
+    }
+
+    // Try as multiple setting
+    if (CLuaArguments result; g_pGame->GetConfig()->GetSettingTable(name, &result))
+    {
+        return result;
+    }
+
+    return false;
+}
+
+bool CLuaGenericDefs::SetServerPassword(std::optional<std::string> rawPassword)
+{
+    std::string password = (rawPassword.has_value() ? std::move(rawPassword.value()) : std::string());
+
+    if (!CStaticFunctionDefinitions::SetServerPassword(password, true))
+        throw std::invalid_argument("password must be shorter than 32 chars and just contain visible characters");
+
+    return true;
+}
+
+std::variant<std::nullptr_t, std::string_view> CLuaGenericDefs::GetServerPassword()
+{
+    if (g_pGame->GetConfig()->HasPassword())
+    {
+        return std::string_view{g_pGame->GetConfig()->GetPassword()};
+    }
+
+    return nullptr;
+}
+
+unsigned int CLuaGenericDefs::GetServerHttpPort()
+{
+    return g_pGame->GetConfig()->GetHTTPPort();
+}
+
+std::string_view CLuaGenericDefs::GetServerName()
+{
+    return std::string_view{g_pGame->GetConfig()->GetServerName()};
+}
+
+bool CLuaGenericDefs::OutputServerLog(std::string_view message)
+{
+    CLogger::LogPrintf(LOGLEVEL_MEDIUM, "%.*s\n", message.length(), message.data());
+    return true;
+}
+
+bool CLuaGenericDefs::OutputScriptDebugLog(lua_State* luaVM, std::string message, std::optional<unsigned char> rawLevel, std::optional<unsigned char> rawColorR,
+                                           std::optional<unsigned char> rawColorG, std::optional<unsigned char> rawColorB)
+{
+    unsigned char level = rawLevel.value_or(3);
+
+    if (level > 4)
+        throw std::invalid_argument("level must be between 0 and 4");
+
+    unsigned char colorR = rawColorR.value_or(255);
+    unsigned char colorG = rawColorG.value_or(255);
+    unsigned char colorB = rawColorB.value_or(255);
+
+    switch (level)
+    {
+        case 0:
+            m_pScriptDebugging->LogDebug(luaVM, colorR, colorG, colorB, "%.*s", message.length(), message.c_str());
+            break;
+        case 1:
+            m_pScriptDebugging->LogError(luaVM, "%.*s", message.length(), message.c_str());
+            break;
+        case 2:
+            m_pScriptDebugging->LogWarning(luaVM, "%.*s", message.length(), message.c_str());
+            break;
+        case 3:
+            m_pScriptDebugging->LogInformation(luaVM, "%.*s", message.length(), message.c_str());
+            break;
+        case 4:
+            m_pScriptDebugging->LogCustom(luaVM, colorR, colorG, colorB, "%.*s", message.length(), message.c_str());
+            break;
+    }
+
+    return true;
+}
+
+bool CLuaGenericDefs::OutputConsole(std::string message, std::optional<CElement*> rawElement)
+{
+    CStaticFunctionDefinitions::OutputConsole(message.c_str(), rawElement.value_or(m_pRootElement));
+    return true;
+}
+
+bool CLuaGenericDefs::OutputChatBox(lua_State* luaVM, std::string message, std::optional<std::variant<CElement*, std::vector<CPlayer*>>> rawVisibleTo,
```

Comment:
Let's use `std::string_view` here. Lua's string are always null terminated. What do you think?

Reply:
I might be wrong, but string-convertible values won't work with `std::string_view` - needs testing.

Reply:
What do you mean?